### PR TITLE
maintain only one instance of the sources table

### LINKED ISSUE #887: Bug: Inconsistency between supported formats
### Where's the bug?

https://www.risingwave.dev/docs/current/sql-create-source/

### Describe the bug

The content of the following two pages overlaps but are also slightly out of sync:
* https://www.risingwave.dev/docs/current/sql-create-source/
* https://www.risingwave.dev/docs/current/data-ingestion/

Differences in Supported sources table:
* Only in Ecosystem page: CDC via Kafka, Amazon S3
* Error in Reference page: PostgreSQL CDC only allows Debezium JSON. It cannot be Maxwell JSON.

Differences in Supported formats list:
* Both page: Avro, JSON, Protobuf, Debezium JSON, Maxwell JSON
* Only in Reference page: Upsert JSON, Upsert AVRO
* Only in Ecosystem page: Canal JSON

> at first there was only one table and it was referenced in the other topic, but another instance got created while resolving conflicts

from [slack thread](https://risingwave-labs.slack.com/archives/C034XCYJPSN/p1683780657693159?thread_ts=1683779995.395109&cid=C034XCYJPSN).

### PR DESCRIPTION
**Pull request by hengm3467**
## Info
- **Description**: 
NOTE: Re-edit the topics by ensuring the texts are copied from the same versions (between two topics for upcoming, and two topics for 0.18, but not across-versions).

Remove the sections on supported sources and formats in the "Overview of sources" topic, and update the corresponding sections in the "SQL CREATE SOURCE" topic. We observe that most users would access the "CREATE SOURCE" page to look for this information, hence the change and keeping the source of truth in the "CREATE SOURCE" topic.

- **Preview**: 
https://pr-888.d2fbku9n2b6wde.amplifyapp.com/docs/upcoming/sql-create-source/
https://pr-888.d2fbku9n2b6wde.amplifyapp.com/docs/upcoming/data-ingestion/

- **Related code PR**: 
[ Provide the link to the code PR here. Delete this part if there's no code PR related. ]

- **Related doc issue**: 
Resolves #887

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

### CODE REVIEW

File: docs/sql/commands/sql-create-source.md [hunk 1]
```diff
@@ -150,7 +165,7 @@ Syntax:
 ROW FORMAT JSON
 ```
 
-For data in Protobuf format, you must specify a message and a schema location. The schema location can be an actual Web location that is in `http://...`, `https://...`, or `S3://...` format. For Kafka data in Protobuf, instead of providing a schema location, you can provide a Confluent Schema Regsitry that RisingWave can get the schema from. For more details about using Schema Registry for Kafka data, see [Read schema from Schema Registry](/create-source/create-source-kafka.md#read-schemas-from-schema-registry).
+For data in Protobuf format, you must specify a message and a schema location. The schema location can be an actual Web location that is in `http://...`, `https://...`, or `S3://...` format. For Kafka data in Protobuf, instead of providing a schema location, you can provide a Confluent Schema Regsitry that RisingWave can get the schema from. For more details about using Schema Registry for Kafka data, see [Read schema from Schema Registry](create-source/create-source-kafka.md#read-schemas-from-schema-registry).
```
Comment (xiangjinwu) [inline docs/sql/commands/sql-create-source.md]: Missing a section header `### Protobuf` (another difference I failed to include in the issue.)

File: versioned_docs/version-0.18.0/sql/commands/sql-create-source.md [hunk 2]
```diff
@@ -24,6 +24,12 @@ ROW FORMAT data_format
 [ ROW SCHEMA LOCATION [ 'location' | CONFLUENT SCHEMA REGISTRY 'schema_registry_url' ] ];
 ```
 
+For the syntax of `schema_definition`, see [Parameters](sql-create-table.md#parameters) in `CREATE TABLE`.
+
+:::note
+To know when a data record is loaded to RisingWave, you can define a column that is generated based on the processing time (`<column_name> timestampz AS proctime()`) when creating the table or source.
```
Comment (xiangjinwu) [inline versioned_docs/version-0.18.0/sql/commands/sql-create-source.md]: I am not sure if this feature is available in 0.18.0. Maybe avoid such unrelated changes in this PR?

Looks like this file is copied from upcoming version. Shall we make sure the updated content of this page is from 0.18.0's data-ingestion page?

File: docs/sql/commands/sql-create-source.md [hunk 3]
```diff
@@ -101,30 +101,45 @@ Names and unquoted identifiers are case-insensitive. Therefore, you must double-
...
+|---------|---------|---------|
+|[Kafka](create-source/create-source-kafka.md)|3.1.0 or later versions |[Avro](#avro), [JSON](#json), [protobuf](#protobuf), [Debezium JSON](#debezium-json) (M), Debezium AVRO (M), [Maxwell JSON](#maxwell-json) (M), [Canal JSON](#canal-json) (M), Upsert JSON, Upsert AVRO|
+|[Redpanda](create-source/create-source-redpanda.md)|Latest|[Avro](#avro), [JSON](#json), [protobuf](#protobuf) |
+|[Pulsar](create-source/create-source-pulsar.md)| 2.8.0 or later versions|[Avro](#avro), [JSON](#json), [protobuf](#protobuf), [Debezium JSON](#debezium-json) (M), [Maxwell JSON](#maxwell-json) (M), [Canal JSON](#canal-json) (M)|
+|[Astra Streaming](guides/connector-astra-streaming.md)|Latest |[Avro](#avro), [JSON](#json), [protobuf](#protobuf)|  
+|[Kinesis](create-source/create-source-kinesis.md)| Latest| [Avro](#avro), [JSON](#json), [protobuf](#protobuf), [Debezium JSON](#debezium-json) (M), [Maxwell JSON](#maxwell-json) (M), [Canal JSON](#canal-json) (M)|
+|[PostgreSQL CDC](guides/ingest-from-pg-cdc)| 10, 11, 12, 13, 14|[Debezium JSON](#debezium-json) (M)|
+|[MySQL CDC](guides/ingest-from-mysql-cdc)| 5.7, 8.0|[Debezium JSON](#debezium-json) (M)|
```
Comment (xiangjinwu) [inline docs/sql/commands/sql-create-source.md]: Links to PostgreSQL and MySQL seem to be broken.

BTW, it would be good if we can have the ci to catch all invalid link issues.

File: versioned_docs/version-0.18.0/sql/commands/sql-create-source.md (hunk 2 above)
Comment (hengm3467) [reply versioned_docs/version-0.18.0/sql/commands/sql-create-source.md]: @xiangjinwu Good point!

File: docs/sql/commands/sql-create-source.md (hunk 3 above)
Comment (hengm3467) [reply docs/sql/commands/sql-create-source.md]: Good suggestion. I have a local instance of yarn validating the links, but this absolute/relative path may be beyond its capabilities. Let me see if there are link validation actions out there.

File: versioned_docs/version-0.18.0/sql/commands/sql-create-source.md (hunk 2 above)
Comment (xiangjinwu) [reply versioned_docs/version-0.18.0/sql/commands/sql-create-source.md]: I meant, there are other changes carried from the upcoming version rather than 0.18.0, for example the inclusion of google pub/sub, or mentioning Debezium/Maxwell/Canal in the first row for Kafka, maybe more.

Although they may be supported in 0.18.0 and these are actually "fixes", it is still better to keep them a separate PR, and reviewed by someone who is more familiar with them.

Comment (hengm3467) [reply versioned_docs/version-0.18.0/sql/commands/sql-create-source.md]: @xiangjinwu I see your point. I just reverted the changes and copy texts only within the same versions. 

Maybe I should have let Emily handle the changes, as she is the one that has been actively maintaining this section. But since I started, let me finish it.

File: docs/sql/commands/sql-create-source.md [hunk 4]
```diff
@@ -99,25 +99,40 @@ Names and unquoted identifiers are case-insensitive. Therefore, you must double-
...
+|[Kafka](/create-source/create-source-kafka.md)|3.1.0 or later versions |[Avro](#avro), [JSON](#json), [protobuf](#protobuf), [Debezium JSON](#debezium-json) (M), Debezium AVRO (M), [Maxwell JSON](#maxwell-json) (M), [Canal JSON](#canal-json) (M), [Upsert JSON](#upsert-json), [Upsert AVRO](#upsert-avro)|
+|[Redpanda](/create-source/create-source-redpanda.md)|Latest|[Avro](#avro), [JSON](#json), [protobuf](#protobuf) |
+|[Pulsar](/create-source/create-source-pulsar.md)| 2.8.0 or later versions|[Avro](#avro), [JSON](#json), [protobuf](#protobuf), [Debezium JSON](#debezium-json) (M), [Maxwell JSON](#maxwell-json) (M), [Canal JSON](#canal-json) (M)|
+|[Astra Streaming](/guides/connector-astra-streaming.md)|Latest |[Avro](#avro), [JSON](#json), [protobuf](#protobuf)|  
+|[Kinesis](/create-source/create-source-kinesis.md)| Latest| [Avro](#avro), [JSON](#json), [protobuf](#protobuf), [Debezium JSON](#debezium-json) (M), [Maxwell JSON](#maxwell-json) (M), [Canal JSON](#canal-json) (M)|
+|[PostgreSQL CDC](/guides/ingest-from-pg-cdc)| 10, 11, 12, 13, 14|[Debezium JSON](#debezium-json) (M)|
+|[MySQL CDC](/guides/ingest-from-mysql-cdc)| 5.7, 8.0|[Debezium JSON](#debezium-json) (M)|
```
Comment (xiangjinwu) [inline docs/sql/commands/sql-create-source.md]: Broken links to PostgreSQL and MySQL, maybe due to missing `.md` suffix

**Review by xiangjinwu — APPROVED** [review]
rest lgtm

File: docs/sql/commands/sql-create-source.md (hunk 4 above)
Comment (hengm3467) [reply docs/sql/commands/sql-create-source.md]: That explains! Fixing it now.

Comment (xiangjinwu) [reply docs/sql/commands/sql-create-source.md]: Seems still broken 😢 
```diff
- ingest-from-pg-cdc
+ ingest-from-postgres-cdc
```
